sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -146,14 +146,11 @@ private static bool IsOn(this ExpressionSyntax expression, SyntaxKind onKind)
             }
         }
 
-        public static bool IsInNameofCall(this ExpressionSyntax expression, SemanticModel semanticModel) =>
-            expression.Parent != null &&
-            expression.Parent.IsKind(SyntaxKind.Argument) &&
-            expression.Parent.Parent != null &&
-            expression.Parent.Parent.IsKind(SyntaxKind.ArgumentList) &&
-            expression.Parent.Parent.Parent != null &&
-            expression.Parent.Parent.Parent.IsKind(SyntaxKind.InvocationExpression) &&
-            ((InvocationExpressionSyntax)expression.Parent.Parent.Parent).IsNameof(semanticModel);
+        public static bool IsInNameofCall(this ExpressionSyntax expression, SemanticModel semanticModel)
```

Comment:
It would be nice to add tests for helpers when we touch them, because they are covered only as secondary effect and it's not documented what they are intended to do.
Unfortunately for this file, I added `SyntaxHelperTest.cs` file in the JWT PR that is not merged yet, so it would be a hassle to synchronize/rebare&rerebase it 😒

Reply:
It's a good point. I will add some tests after both commits are merged to avoid the merge pain.